extras/create_snapshots_reports_scorecard.py

```diff
@@ -288,66 +287,106 @@ def create_snapshot(db, cyhy_db_section, org_id, use_only_existing_snapshots):
                 logging.info(
-                    " - Includes successful descendant snapshot(s): %s", org_descendants
+                    "[%s]  - Includes successful descendant snapshot(s): %s",
+                    threading.current_thread().name,
+                    org_descendants,
                 )
                 successful_snapshots.extend(org_descendants)
     else:
-        logging.error("Unsuccessful snapshot: %s", org_id)
+        logging.error(
+            "[%s] Unsuccessful snapshot: %s", threading.current_thread().name, org_id
+        )
         with fs_lock:
             failed_snapshots.append(org_id)
             if org_descendants and not use_only_existing_snapshots:
                 logging.error(
-                    " - Unsuccessful descendant snapshot(s): %s", org_descendants,
+                    "[%s]  - Unsuccessful descendant snapshot(s): %s",
+                    threading.current_thread().name,
+                    org_descendants,
                 )
                 failed_snapshots.extend(org_descendants)
-        logging.error("Stderr failure detail: %s %s", data, err)
+        logging.error(
+            "[%s] Stderr failure detail: %s %s",
+            threading.current_thread().name,
+            data,
+            err,
+        )
     return snapshot_process.returncode
 
 
-def create_snapshots_from_list(org_list, db, cyhy_db_section):
-    """Create a snapshot for each organization in a list."""
-    for org_id in org_list:
+def generate_snapshots_from_list(db, cyhy_db_section):
+    """Attempt to generate a snapshot for each organization in a global list.
+
+    Each thread pulls an organization ID from the global list
+    (snapshots_to_generate) and attempts to generate a snapshot for it."""
+    global snapshots_to_generate
+    while True:
+        with stg_lock:
+            logging.debug(
+                "[%s] %d snapshot(s) left to generate",
+                threading.current_thread().name,
+                len(snapshots_to_generate),
+            )
+            if snapshots_to_generate:
+                org_id = snapshots_to_generate.pop(0)
+            else:
+                logging.info(
+                    "[%s] No snapshots left to generate - thread exiting",
+                    threading.current_thread().name,
+                )
+                break
+
         logging.info(
             "[%s] Starting snapshot for: %s", threading.current_thread().name, org_id
         )
-        create_snapshot(db, cyhy_db_section, org_id, use_only_existing_snapshots=False)
+        generate_snapshot(db, cyhy_db_section, org_id, use_only_existing_snapshots=False)
 
 
-def generate_weekly_snapshots(db, cyhy_db_section):
-    """Generate all snapshots needed in order to generate the CyHy reports."""
+def manage_snapshot_threads(db, cyhy_db_section):
+    """Build the lists of reports and snapshots to be generated, then spawn the
+    threads that generate the snapshots."""
```

Comment:
In the same vein as https://github.com/cisagov/cyhy-reports/pull/106#discussion_r1473450650 would you reformat this docstring please.

Reply:
@mcdonnnj Can you please clarify how you would like this one to be reformatted?  The comment you linked to had two sentences and @jsf9k asked me to split them up with a blank line in between.  This pydoc description is a single sentence, so I'm not clear on what you'd like me to do here.  Are you asking me to remove the line break so that the pydoc string is on a single line?  Or something else?

Reply:
After chatting with @jsf9k, we came up with https://github.com/cisagov/cyhy-reports/pull/106/commits/aceb30388e15fd0171e0d6f907b327d66affdfde.  Please resolve this convo if that works for you.

Reply:
Apologies for being unclear with my feedback. That looks perfect and in-line with the intent behind my comment.

Reply:
@mcdonnnj and I are on the same wavelength.  :laughing: